asserts/findwildcard.go

```diff
@@ -101,11 +117,67 @@ func findWildcardDescend(top, current string, descendantWithWildcard []string, f
 			return fmt.Errorf("findWildcard: invoked with malformed wildcard: %v", err)
 		}
 		if ok {
-			err = findWildcardDescend(top, filepath.Join(current, name), descendantWithWildcard[1:], foundCb)
+			err = findWildcardDescend(top, filepath.Join(current, name), descendantWithWildcard[1:], seqnum, foundCb)
 			if err != nil {
 				return err
 			}
 		}
 	}
 	return nil
 }
+
+func findWildcardSequence(top, current, seqWildcard string, descendantWithWildcard []string, seqnum int, foundCb func(relpath []string) error) error {
+	filter := func(i int) bool { return true }
+	if seqnum != -1 {
+		if seqWildcard == "#>" {
+			filter = func(i int) bool { return i > seqnum }
+		} else {
+			filter = func(i int) bool { return i < seqnum }
+		}
+	}
+
+	d, err := os.Open(current)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+	var seq []int
+	for {
+		names, err := d.Readdirnames(100)
```

Comment:
Is there a chance that at the time we read the assertions database there are temporary files in the same directory, that have a random name that would break this logic. Should we filter out some canned pattern (e.g. those that are created by our atomic write functions)

Reply:
so writes of files use AtomicWriteFiles, two things:

* there are never directories and files at the same level
* there's a mutex around everything that stops writing and reading at the same time